src/System.Text.JsonLab/System/Text/Json/JsonWriterUtf8.cs

```diff
@@ -507,15 +507,23 @@ public void WriteValue(ulong value)
         /// <param name="value">The boolean value to be written to JSON data.</param>
         public void WriteValue(bool value)
         {
-            //TODO: Optimize, just like WriteValue(long value)
-            WriteItemSeperatorUtf8();
-            _firstItem = false;
-            WriteSpacingUtf8();
+            ReadOnlySpan<byte> literal = value ? JsonConstants.TrueValue : JsonConstants.FalseValue;
 
-            if (value)
-                WriteJsonValue(JsonConstants.TrueValue);
-            else
-                WriteJsonValue(JsonConstants.FalseValue);
+            int seperatorSize = ItemSeperatorSize();
+            int spacingSize = SpacingSize();
+            int literalSize = LiteralSize(literal);
+
+            int bytesNeeded = seperatorSize + spacingSize + literalSize;
+
+            Span<byte> byteBuffer = EnsureBuffer(bytesNeeded);
+
+            int idx = 0;
+            idx += this.WriteItemSeperator(byteBuffer.Slice(idx, seperatorSize));
```

Comment:
Can we do all these in a single method to avoid the multiple span slices and additions to idx?